packages/databyss-ui/Editor/EditorPage.js

```diff
@@ -40,18 +42,27 @@ const EditorPage = ({ children }) => {
   }
 
   const onBlockBlur = (id, text, editableState) => {
-    // check if block is empty
     // if empty replace block with virgin block
     if (text.length === 0 && id) {
-      // check block to see if ranges exist
-      dispatchEditor(clearBlock(id, editableState))
+      if (!isBlockEmpty(id, editableState)) {
```

Comment:
I don't understand this case. How can `text.length` be 0 but block not empty?

Reply:
this dispatches a helper function which will replace the current block with a new block. blocks can have left over `marks` or have a `type` which isn't `ENTRY`